src/argilla/client/feedback/training/schemas.py

```diff
@@ -154,49 +159,74 @@ def for_text_classification(
+            ...     values = [annotation["value"] for annotation in sample["label"]]
+            ...     counter = Counter(values)
+            ...     if counter:
+            ...         most_common = counter.most_common()
+            ...         max_frequency = most_common[0][1]
+            ...         most_common_elements = [element for element, frequency in most_common if frequency == max_frequency]
+            ...         label = random.choice(most_common_elements)
+            ...         return (text, label)
+            ...     else:
+            ...         return None
+            >>> task = TrainingTask.for_text_classification(formatting_func=formatting_func)
+            >>> dataset.prepare_for_training(framework="...", task=task)
         """
-        if isinstance(
-            label,
-            (
-                LabelQuestionUnification,
-                MultiLabelQuestionUnification,
-                RatingQuestionUnification,
-                RankingQuestionUnification,
-            ),
-        ):
-            if label_strategy is not None:
-                raise ValueError("label_strategy is already defined via Unification class.")
+        if text or label:
+            if formatting_func is not None:
+                raise ValueError("`formatting_func` is already defined, so you cannot define `text` and `label`.")
+        elif text or label:
+            raise ValueError("You must define both `text` and `label`.")
+        elif not (text or label) and formatting_func is None:
+            raise ValueError("You must define either (`text` and `label`), or a `formatting_func`.")
+
+        if formatting_func is not None:
+            if text or label:
+                raise ValueError("`formatting_func` is already defined, so you cannot define `text` and `label`.")
+            return TrainingTaskForTextClassification(formatting_func=formatting_func)
         else:
-            unification_kwargs = {"question": label}
-            if label_strategy is not None:
-                unification_kwargs["strategy"] = label_strategy
-            else:
-                _LOGGER.info(f"No label strategy defined. Using default strategy for {type(label)}.")
-            if isinstance(label, RatingQuestion):
-                label = RatingQuestionUnification(**unification_kwargs)
-            elif isinstance(label, MultiLabelQuestion):
-                label = MultiLabelQuestionUnification(**unification_kwargs)
-            elif isinstance(label, LabelQuestion):
-                label = LabelQuestionUnification(**unification_kwargs)
-            elif isinstance(label, RankingQuestion):
-                label = RankingQuestionUnification(**unification_kwargs)
+            if isinstance(
+                label,
+                (
+                    LabelQuestionUnification,
+                    MultiLabelQuestionUnification,
+                    RatingQuestionUnification,
+                    RankingQuestionUnification,
```

Comment:
Maybe we can defined the supported unification strategies as an `Union` in a constant variable and just unwrap it for the `isinstance` check